Fat JAR broken (breaks Lucene)

The `./gradlew jar` task makes a jar which is somehow missing some info lucene needs in META-INF to do some reflection (or something?! see https://stackoverflow.com/a/42462012 )

For example I get the following when trying to run cineast setup against it when it gets to a Lucene based index:

```
2020-07-23 11:15:36 ERROR CottonDDLService:260 - Error while creating index 'cineast.cineast_tags.index-lucene-cineast_cineast_tags_name'
java.util.ServiceConfigurationError: Cannot instantiate SPI class: org.apache.lucene.codecs.lucene70.Lucene70Codec
	at org.apache.lucene.util.NamedSPILoader.reload(NamedSPILoader.java:82)
	at org.apache.lucene.util.NamedSPILoader.<init>(NamedSPILoader.java:51)
	at org.apache.lucene.util.NamedSPILoader.<init>(NamedSPILoader.java:38)
	at org.apache.lucene.codecs.Codec$Holder.<clinit>(Codec.java:47)
	at org.apache.lucene.codecs.Codec.getDefault(Codec.java:143)
	at org.apache.lucene.index.LiveIndexWriterConfig.<init>(LiveIndexWriterConfig.java:129)
	at org.apache.lucene.index.IndexWriterConfig.<init>(IndexWriterConfig.java:157)
	at org.vitrivr.cottontail.database.index.lucene.LuceneIndex.<init>(LuceneIndex.kt:92)
	at org.vitrivr.cottontail.database.index.IndexType.create(IndexType.kt:48)
	at org.vitrivr.cottontail.database.entity.Entity.createIndex(Entity.kt:190)
	at org.vitrivr.cottontail.server.grpc.services.CottonDDLService.createIndex(CottonDDLService.kt:237)
	at org.vitrivr.cottontail.grpc.CottonDDLGrpc$MethodHandlers.invoke(CottonDDLGrpc.java:1020)
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:172)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331)
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:818)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.lang.IllegalArgumentException: An SPI class of type org.apache.lucene.codecs.PostingsFormat with name 'Lucene50' does not exist.  You need to add the corresponding JAR file supporting this SPI to your classpath.  The current classpath supports the following names: [IDVersion]
	at org.apache.lucene.util.NamedSPILoader.lookup(NamedSPILoader.java:116)
	at org.apache.lucene.codecs.PostingsFormat.forName(PostingsFormat.java:112)
	at org.apache.lucene.codecs.lucene70.Lucene70Codec.<init>(Lucene70Codec.java:166)
	at org.apache.lucene.codecs.lucene70.Lucene70Codec.<init>(Lucene70Codec.java:81)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:124)
	at java.base/jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:346)
	at java.base/java.lang.Class.newInstance(Class.java:604)
	at org.apache.lucene.util.NamedSPILoader.reload(NamedSPILoader.java:72)
	... 19 more
```

It works just fine when doing `./gradlew run` (but then the REPL doesn't work properly)

--- a/build.gradle
+++ b/build.gradle
@@ -6,16 +6,18 @@ buildscript {
     ext.kotlinVersion = '1.3.60'
 
     repositories {
-        mavenCentral()
+        jcenter()
     }
     dependencies {
         classpath "com.google.protobuf:protobuf-gradle-plugin:0.8.8"
         classpath "org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion"
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
         classpath "org.jetbrains.kotlin:kotlin-serialization:$kotlinVersion"
+        classpath 'com.github.jengelman.gradle.plugins:shadow:6.0.0'
     }
 }
 
+apply plugin: 'com.github.johnrengelman.shadow'
 apply plugin: 'application'
 apply plugin: 'java'
 apply plugin: 'kotlin'
@@ -117,18 +119,8 @@ dependencies {
     compile group: 'org.apache.commons', name: 'commons-math3', version: '3.6.1'
 }
 
-jar {
-
-    doFirst {
-        // FAT JAR (with dependencies)
-        from { configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) } }
-    }
-    // To prevent unwanted files
-    exclude 'META-INF/*.RSA', 'META-INF/*.SF', 'META-INF/*.DSA'
-
-    manifest {
-        attributes 'Main-Class': mainClassName
-    }
+shadowJar {
+    mergeServiceFiles()
 }
 
 idea {

